src/modeler/AutoRest.Swagger/Validation/CollectionObjectPropertiesNamingValidation.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using AutoRest.Core.Logging;
+using AutoRest.Core.Properties;
+using AutoRest.Core.Validation;
+using AutoRest.Swagger.Model;
+using AutoRest.Swagger.Model.Utilities;
+using System.Linq;
+
+namespace AutoRest.Swagger.Validation
+{
+    public class CollectionObjectPropertiesNamingValidation : TypedRule<Dictionary<string, Dictionary<string, Operation>>>
+    {
+        private readonly Regex ListRegex = new Regex(@".+_List([^_]*)$", RegexOptions.IgnoreCase);
+        public override IEnumerable<ValidationMessage> GetValidationMessages(Dictionary<string, Dictionary<string, Operation>> entity, RuleContext context)
+        {
+            // get all operation objects that are either of get or post type
+            var serviceDefinition = (ServiceDefinition)context.Root;
```

Comment:
minor: in general I'd prefer concrete type over `var` because that helps in understanding the type :)